pkg/bdb/generic.go

```diff
@@ -34,7 +34,8 @@ func marshal[T any, M Message[T]](t M) ([]byte, error) {
 func unmarshal[T any, M Message[T]](b []byte) (M, error) {
 	var t T
 
-	if err := unmarshalOpts.Unmarshal(b, any(&t).(proto.Message)); err != nil {
+	msg, _ := any(&t).(proto.Message)
```

Comment:
I _think_ you should be able to write this without a type assertion using:
```go
msg := M(&t)
```

Because `M` is supported to be a `*T` and a `proto.Message` so it should convert cleanly.

Reply:
Thanks!